datafusion-common/src/dfschema.rs

```diff
@@ -281,6 +281,14 @@ impl DFSchema {
             .all(|(dffield, arrowfield)| dffield.name() == arrowfield.name())
     }
 
+    /// Check to see if df field data types in df schema matches with input df schema
+    pub fn matches_arrow_schema_with_type(&self, df_schema: &DFSchema) -> bool {
+        self.fields
+            .iter()
```

Comment:
in postgresql union support  type cast like below
```
postgres=# select '1.2' as a union select 2.3 as b union select 3 as b ;
  a
-----
 2.3
   3
 1.2
(3 rows)

postgres=# \gdesc
 Column |  Type
--------+---------
 a      | numeric
(1 row)
```

Maybe we can open a ticket support this ?

Reply:
@Ted-Jiang Agree with that.
Maybe we can solve `Alias` issue first, than open a new issue to fully discuss data type compatible cast?